### DISCUSSION TITLE
Access the WebPartContext in a web component of a extensibility libraray

### DISCUSSION THREAD

**Discussion by MarkusLanger** [microsoft-search/pnp-modern-search#3482] [Q&A] [answered]
Hello,
I need the WebPartContext in a custom web component. Unfortunately, I have not found a way to access it.
In a layout it is possible to access the WebPartContext by "super.context".
Is it also possible in a web component?
If it is currently not possible, what do I have to change to make it work?

Thanks a lot,
Markus

**Comment by patrikhellgren** [comment, marked as answer]
See my comment [here](https://github.com/microsoft-search/pnp-modern-search/discussions/2547#discussioncomment-6058122) on how to pass BaseComponentContext to a web component. If you really need WebPartContext you probably could use the same method.
I also just added a sample of this to the [microsoft-search/pnp-modern-search-extensibility-samples](https://github.com/microsoft-search/pnp-modern-search-extensibility-samples) repo. You can find the PR [here](https://github.com/microsoft-search/pnp-modern-search-extensibility-samples/pull/19).
